### PR TITLE
[DO NOT MERGE] #300 Missing documentation: Rancher monitoring requires port 10254

### LINKED ISSUE #300: Missing Documentation: Rancher Monitoring requires port 10254
Rancher Monitoring requires that the host firewall be open on port 10254. Prometheus will scrape metrics from this port. However, this port is not correctly documented in the Rancher Docs.

There are two places where this could be documented:

1. https://docs.ranchermanager.rancher.io/pages-for-subheaders/monitoring-and-alerting , which does not mention port 10254.

2. The Rancher documentation at [Ports for Rancher Launched Kubernetes Clusters using Custom Nodes](https://docs.ranchermanager.rancher.io/v2.5/getting-started/installation-and-upgrade/installation-requirements/port-requirements#ports-for-rancher-launched-kubernetes-clusters-using-custom-nodes) uses a grid to explain the firewall requirements. In the following section, I'm looking at the "From Worker Plane Nodes", "To Worker Plane Nodes". Note footnote 4 in the box labeled "10254 TCP (4)".

image

Footnote 4 says:

> 4. Local traffic to the node itself (not across nodes).

Users have pointed out this discrepancy in the docs in a couple previous issues, such as:
* https://github.com/rancher/rancher-docs/issues/136#issuecomment-1247331477

### PR DESCRIPTION
**Pull request by martyav**
Resolves #300 

<s>I had a nitpicky question about how to update a footnote, which is why this is marked as a draft for now. </s>

I added information about the port to the Enable Monitoring page, edited a footnote in a table, and cleaned up some obvious style problems in the Monitoring and Alerting page.

### CODE REVIEW

**Comment by martyav** [comment]
Got confirmation on my question to @stefanlasiewski

File: docs/pages-for-subheaders/monitoring-and-alerting.md [hunk 1]
```diff
@@ -14,18 +14,16 @@ By viewing data that Prometheus scrapes from your cluster control plane, nodes,
...
-- Monitor the state and processes of your cluster nodes, Kubernetes components, and software deployments
-- Define alerts based on metrics collected via Prometheus
-- Create custom Grafana dashboards
-- Configure alert-based notifications via Email, Slack, PagerDuty, etc. using Prometheus Alertmanager
-- Defines precomputed, frequently needed or computationally expensive expressions as new time series based on metrics collected via Prometheus
-- Expose collected metrics from Prometheus to the Kubernetes Custom Metrics API via Prometheus Adapter for use in HPA
+- Monitors the state and processes of your cluster nodes, Kubernetes components, and software deployments.
+- Defines alerts based on metrics collected via Prometheus.
+- Creates custom Grafana dashboards.
+- Configures alert-based notifications via Email, Slack, PagerDuty, etc. using Prometheus Alertmanager.
```
Comment (martyav) [inline docs/pages-for-subheaders/monitoring-and-alerting.md:22]:
```suggestion
- Configures alert-based notifications via email, Slack, PagerDuty, etc. using Prometheus Alertmanager.
```

File: docs/how-to-guides/advanced-user-guides/monitoring-alerting-guides/enable-monitoring.md [hunk 2]
```diff
@@ -10,10 +10,11 @@ You can enable monitoring with or without SSL.
...
-- When installing monitoring on an RKE cluster using RancherOS or Flatcar Linux nodes, change the etcd node certificate directory to `/opt/rke/etc/kubernetes/ssl`.
-- For clusters provisioned with the RKE CLI and the address is set to a hostname instead of an IP address, set `rkeEtcd.clients.useLocalhost` to `true` during the Values configuration step of the installation. The YAML snippet will look like the following:
+- Allow traffic on ports 9796 and 10254 for each of your nodes. Prometheus scrapes metrics from these ports.
+- Make sure that your cluster fulfills the resource requirements. The cluster should have at least 1950Mi memory available, 2700m CPU, and 50Gi storage. See [Configuring Resource Limits and Requests
+](../../../reference-guides/monitoring-v2-configuration/helm-chart-options.md#configuring-resource-limits-and-requests) for a breakdown of the resource limits and requests.
```
Comment (martyav) [inline docs/how-to-guides/advanced-user-guides/monitoring-alerting-guides/enable-monitoring.md:15]:
```suggestion
- Make sure that your cluster fulfills the resource requirements. The cluster should have at least 1950Mi memory available, 2700m CPU, and 50Gi storage. See [Configuring Resource Limits and Requests](../../../reference-guides/monitoring-v2-configuration/helm-chart-options.md#configuring-resource-limits-and-requests) for a breakdown of the resource limits and requests.
```

**Review by geethub97 — APPROVED** [review]
LGTM

**Review by btat — APPROVED** [review]

**Comment by martyav** [comment]
While discussing this PR with Geet and Arvind, it was pointed out that manual port opening is only relevant to older versions of Rancher that have V1 monitoring. Current versions use PushProx with V2 monitoring as described here: https://github.com/rancher/prometheus-federator/blob/main/docs/developer/monitoring_v2.md#pushprox-exporters-for-most-kubernetes-internal-components

Furthermore, docs for older versions of Rancher already note the 10254 port requirement: https://github.com/rancher/docs/pull/3082/files

However, the investigation on this issue suggests that we need to refresh our monitoring pages to clearly mention PushProx in regards to V2.

**Comment by stefanlasiewski** [comment]
It's been a while since I filed the request, but I believe that my request in https://github.com/rancher/rancher-docs/issues/300 was in regards to Rancher Monitoring v2. I have had the same problem on two different Rancher 2.6 clusters, and one Rancher 2.5 cluster running Monitoring v2.

**Comment by martyav** [comment]
@stefanlasiewski Looks like it can still be an issue with v2 if you have PushProx disabled or upgraded Rancher from a previous version that had v1 monitoring

**Comment by stefanlasiewski** [comment]
> upgraded Rancher from a previous version that had v1 monitoring

Good to know! That certainly applies to 1-2 of my clusters. I'll keep that in mind.

**Review by geethub97 — APPROVED** [review]

**Pull request merged**